From 2047c9fe37ab3850066d1f41cc2e72a9c9a8c1ec Mon Sep 17 00:00:00 2001
From: Sergey Volkov <Sergey.Volkov@promwad.com>
Date: Mon, 7 Feb 2022 19:35:39 +0300
Subject: [PATCH 2/8] try boot from both SD/MMC slots on RZX-27

external MMC first
---
 src/config-rzx27.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/config-rzx27.h b/src/config-rzx27.h
index c4d2f22..3fe9486 100644
--- a/src/config-rzx27.h
+++ b/src/config-rzx27.h
@@ -16,5 +16,6 @@
 /* MMC parameters */
 #define MMC_ID          1
 #define MMC_1BIT        1
+#define MMC_ID2         0
 #define PAGE_SIZE       512 /* 512, 2048 or 4096 */
 #define PAGE_PER_BLOCK  1
-- 
2.35.1

