From 1e38397a3cf84d0bf839012c78722cb50421bf0a Mon Sep 17 00:00:00 2001
From: Sergey Volkov <Sergey.Volkov@promwad.com>
Date: Sat, 13 Feb 2021 17:45:44 +0300
Subject: [PATCH 4/7] allow use ubiboot with MBR

---
 ldscripts/target-jz4740.ld | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/ldscripts/target-jz4740.ld b/ldscripts/target-jz4740.ld
index 333ab94..fc666f9 100644
--- a/ldscripts/target-jz4740.ld
+++ b/ldscripts/target-jz4740.ld
@@ -3,14 +3,18 @@ ENTRY(_start)
 
 MEMORY
 {
-	low8k	: ORIGIN = 0x80000000 , LENGTH = 0x2000
+	pre_mbr	: ORIGIN = 0x80000000 , LENGTH =    440
+	low8k	: ORIGIN = 0x80000200 , LENGTH = 0x1E00
 	high8k	: ORIGIN = 0x80002000 , LENGTH = 0x2000
 }
 
 SECTIONS
 {
-	.text : {
+	.startup : {
 		*(.text.1)
+	} > pre_mbr
+
+	.text : {
 		*(.text.0)
 	} > low8k
 
-- 
2.13.6

