From 2d8af3715d0d5ff94298465302718238b9741c24 Mon Sep 17 00:00:00 2001
From: Sergey Volkov <Sergey.Volkov@promwad.com>
Date: Mon, 7 Feb 2022 21:30:26 +0300
Subject: [PATCH 6/8] fix extra warnings

---
 Makefile     | 4 ++--
 src/serial.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index dbc35ee..74256cb 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ OBJCOPY = $(CROSS_COMPILE)objcopy
 OBJDUMP = $(CROSS_COMPILE)objdump
 NM = $(CROSS_COMPILE)nm
 
-CFLAGS	:= -Wall -Os -fno-pic -mno-abicalls -mno-check-zero-division -ffreestanding -flto
+CFLAGS	:= -Wall -Wextra -Os -fno-pic -mno-abicalls -mno-check-zero-division -ffreestanding -flto
 CFLAGS	+= $(CFLAGS_all)
 CPPFLAGS := -DBOARD_$(BOARD) -DJZ_VERSION=$(JZ_VERSION)
 LDFLAGS := -nostdlib -EL
@@ -41,7 +41,7 @@ OUTDIR	:= output/$(CONFIG)
 OBJS	:= utils.o mmc.o fat.o head.o uimage.o
 
 ifdef GC_FUNCTIONS
-	CFLAGS += -ffunction-sections
+	CFLAGS += -ffunction-sections -fdata-sections
 	LDFLAGS += -Wl,--gc-sections
 endif
 
diff --git a/src/serial.c b/src/serial.c
index b31583b..b0b4cf4 100644
--- a/src/serial.c
+++ b/src/serial.c
@@ -119,7 +119,7 @@ void serial_puti(unsigned int value)
 		}
 	} else {
 		message[0] = 0;
-		i++;
+		i = 1;
 	}
 
 	for (; i; i--)
-- 
2.35.1

